extension/src/setup/index.ts

```diff
@@ -161,7 +136,16 @@ export class Setup
   public async getCliVersion(cwd: string, tryGlobalCli?: true) {
     await this.config.isReady()
     try {
-      return await this.dvcReader.version(cwd, tryGlobalCli)
+      if (tryGlobalCli) {
+        return await this.internalCommands.executeCommand(
+          AvailableCommands.GLOBAL_VERSION,
+          cwd
+        )
+      }
+      return await this.internalCommands.executeCommand(
+        AvailableCommands.VERSION,
+        cwd
+      )
```

Comment:
[F] Move the logic out of the "dumb" reader and into here. Wrote some tests to cover it.